ironfish/src/indexers/minedBlocksIndexer.ts

```diff
@@ -270,6 +274,37 @@ export class MinedBlocksIndexer {
     await this.meta.put('headHash', hashString, tx)
   }
 
+  async removeMinedBlocks(accountName: string): Promise<void> {
+    await this.meta.put('accountToRemove', accountName)
+
+    const hashes: Buffer[] = []
+    const sequences: number[] = []
+
+    await this.database.transaction(async (tx) => {
```

Comment:
We probably only need a transaction around the operations and not loading the entire database for pruning. Transactions cache all read state, so this will keep everything loaded in memory because of how this works. It also holds a really long lock.